exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/transactions/TransferTxTest.java

```diff
@@ -142,45 +180,50 @@ void executeTransfer() throws CloseFailuresException {
 
   @Test
   @RequiresNativeLibrary
-  void executeTransferToTheSameWallet() throws CloseFailuresException {
+  void executeTransfer_NoSuchFromWallet() throws CloseFailuresException {
     try (Database db = MemoryDb.newInstance();
          Cleaner cleaner = new Cleaner()) {
       Fork view = db.createFork(cleaner);
+      // Create a receiver’s wallet with the given initial balance
+      long initialBalance = 50L;
+      createWallet(view, toKey, initialBalance);
 
-      long initialBalance = 100L;
-      createWallet(view, fromKey, initialBalance);
-
-      // Create and execute the transaction
       long seed = 1L;
-      long transferSum = 40L;
-      TransferTx tx = withMockMessage(seed, fromKey, fromKey, transferSum);
-      tx.execute(view);
 
-      // Check that the balance of the wallet remains the same
+      long transferValue = 50L;
+      TransferTx tx = withMockMessage(seed, fromKey, toKey, transferValue);
+      // Execute the transaction that attempts to transfer from an unknown wallet
+      TransactionExecutionException e = assertThrows(
+          TransactionExecutionException.class, () -> tx.execute(view));
+      assertThat(e.getErrorCode(), equalTo(UNKNOWN_SENDER.errorCode));
+
+      // Check that balance of fromKey is unchanged
       CryptocurrencySchema schema = new CryptocurrencySchema(view);
-      ProofMapIndexProxy<PublicKey, Wallet> wallets = schema.wallets();
-      assertThat(wallets.get(fromKey).getBalance(), equalTo(initialBalance));
+      MapIndex<PublicKey, Wallet> wallets = schema.wallets();
+      assertThat(wallets.get(toKey).getBalance(), equalTo(initialBalance));
     }
   }
 
   @Test
   @RequiresNativeLibrary
-  void executeNoSuchFromWallet() throws CloseFailuresException {
+  void executeTransfer_NoSuchToWallet() throws CloseFailuresException {
     try (Database db = MemoryDb.newInstance();
          Cleaner cleaner = new Cleaner()) {
       Fork view = db.createFork(cleaner);
-      // Create source wallet with the given initial balance
-      long initialBalance = 50L;
+      // Create a sender’s wallet
+      long initialBalance = 100L;
       createWallet(view, fromKey, initialBalance);
 
+      // Create and execute the transaction that attempts to transfer to unknown wallet
+      long transferValue = 50L;
       long seed = 1L;
 
-      long transferValue = 50L;
       TransferTx tx = withMockMessage(seed, fromKey, toKey, transferValue);
-      // Execute the transaction that attempts to transfer to an unknown wallet
-      tx.execute(view);
+      TransactionExecutionException e = assertThrows(
+          TransactionExecutionException.class, () -> tx.execute(view));
+      assertThat(e.getErrorCode(), equalTo(UNKNOWN_RECEIVER.errorCode));
 
-      // Check that balance of fromKey is unchanged
+      // Check that balance of toKey is unchanged
```

Comment:
? The code does not correspond to this comment

Reply:
:+1: , I'll remove these no longer needed (with exceptions) methods.